save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/projectcollection/CollectionView.kt

```diff
@@ -32,14 +54,22 @@ external interface CreationViewProps : Props {
     var currentUserInfo: UserInfo?
 }
 
+/**
+ * [State] of Collection view component
+ */
+external interface CollectionViewState : State, HasSelectedMenu<ProjectTab>
+
 /**
  * A view with collection of projects
  */
 @JsExport
 @OptIn(ExperimentalJsExport::class)
-class CollectionView : AbstractView<CreationViewProps, State>() {
-    @Suppress("MAGIC_NUMBER", "TYPE_ALIAS")
-    private val projectsTable: FC<TableProps<ProjectDto>> = tableComponent(
+@Suppress(
+    "TYPE_ALIAS",
+    "MAGIC_NUMBER",
+)
+class CollectionView : AbstractView<CreationViewProps, CollectionViewState>() {
```

Comment:
One day we'll forget about class components

Reply:
Yes. But it will be in some other PR.

Reply:
Sure